container-disc/src/main/java/com/yahoo/container/jdisc/component/Deconstructor.java

```diff
@@ -28,69 +28,76 @@
     private final ScheduledExecutorService executor =
             Executors.newScheduledThreadPool(1, ThreadFactoryFactory.getThreadFactory("deconstructor"));
 
-    private final int delay;
+    private final Duration delay;
 
     public Deconstructor(boolean delayDeconstruction) {
-        delay = delayDeconstruction ? 60 : 0;
+        this.delay = delayDeconstruction ? Duration.ofSeconds(60) : Duration.ZERO;
     }
 
 
     @Override
     public void deconstruct(Object component) {
+        // The mix of deconstructing some components on dedicated thread and some on caller thread could in theory violate
```

Comment:
I suggest to delete this comment since the better comment is further down. Let's fix the issue instead in another PR. I don't really understand what you mean by 'ordering constraints' either.

Reply:
Components should be destructed in correct order in case the destructor uses other components injected in constructor. Anyway, I'll remove the comment.